### PR TITLE
front: add some data sources and licenses

### LINKED ISSUE #7621: Add data sources and licenses
### Who would benefit from this feature?

End users

### What is this feature about?

Add a section in "About" crediting the authors and sources of open data used and the licenses of use.
- Terrain https://registry.opendata.aws/terrain-tiles/
- IGN https://geoservices.ign.fr/services-web-experts-ortho
- Train pictures http://www.mlgtraffic.net/index_fr.html 
- Open data SNCF for train import 

### Why is this feature valuable?

Open source good practice & respecting regulations

### Anything else people should know?

... has to be taken into account

### PR DESCRIPTION
**Pull request by clarani**
closes #7621

- add some data sources and licenses
- refacto the licenseAttribution component to separe the data sources and the licenses

image

### CODE REVIEW

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 1]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
-
-  // mglTraffic is not a lib from package.json, so we add it manually
-  const mglTrafficAttribution: attributionsLicensesType = {
-    licenses: 'CC-BY-NC-SA 3.0',
+// Libs which are not in the package.json, so we add it statically
+const DATA_SOURCES: AttributionLicense[] = [
+  {
+    license: '',
+    publisher: 'IGN',
+    copyright: '',
```
Comment (Yohh) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: I can see a copyright in the website

image

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 2]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
+    version: '',
+    name: 'IGN Ortho',
+    url: 'https://geoservices.ign.fr/services-web-experts-ortho',
+  },
+  {
+    license: '',
```
Comment (Yohh) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: there is a license: https://data.sncf.com/pages/licence/#A1

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 3]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
-const LicenseAttributions = () => {
-  const { t } = useTranslation('home/navbar');
-
-  // mglTraffic is not a lib from package.json, so we add it manually
-  const mglTrafficAttribution: attributionsLicensesType = {
-    licenses: 'CC-BY-NC-SA 3.0',
+// Libs which are not in the package.json, so we add it statically
+const DATA_SOURCES: AttributionLicense[] = [
+  {
+    license: '',
```
Comment (Yohh) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: it seems there is a license: https://geoservices.ign.fr/cgu-licences

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 4]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
+    license: 'CC-BY-NC-SA 3.0',
     publisher: 'Marc Le Gad',
     copyright: '',
     version: '',
     name: 'MLG Traffic',
     url: 'http://www.mlgtraffic.net',
-  };
+  },
+  {
+    license: '',
```
Comment (Yohh) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: same [here](https://github.com/tilezen/joerd/blob/master/docs/attribution.md): 
image

**Review by Yohh — COMMENTED** [review]
nice!

(this old component obviously did need a refacto 😉)

a few comments that seems important

**Comment by Yohh** [comment]
I wonder if we could/should add a copy of the licenses in the project

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 1 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 2 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 3 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 4 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

**Review by Yohh — APPROVED** [review]
LGTM

> [!NOTE]
> we should consider my last question, even if it can be totally unnecessary

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 5]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
+    url: 'https://registry.opendata.aws/terrain-tiles/',
+  },
+];
+
+type LicenseCardProps = { attribution: AttributionLicense };
+
+const LicenseCard = ({
+  attribution: { name, version, copyright, publisher, url, license },
+}: LicenseCardProps) => (
+  <a key={name} href={url} rel="noreferrer">
```
Comment (SharglutDev) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: I think it's better this way so we don't have to go to previous page to go back in the app :)

```suggestion
  <a key={name} href={url} rel="noreferrer" target="blank">
```

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 6]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
+    <h3 className="d-flex mr-1 mb-0">
+      {name}
+      <small className="d-flex align-items-center ml-2">
+        {`${version}`}
```
Comment (SharglutDev) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Should be enough

```suggestion
        {version}
```

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 5 above)
Comment (SharglutDev) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx]: There is a warning in console 

Capture d’écran 2024-09-19 à 09 18 44

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx [hunk 7]
```diff
@@ -2,57 +2,95 @@ import { useTranslation } from 'react-i18next';
...
+const LicenseAttributions = () => {
+  const { t } = useTranslation('home/navbar');
 
-  const attributionsArray = [
-    ...(Object.values(attributionsLicenses) as attributionsLicensesType[]),
-    mglTrafficAttribution,
-  ];
+  const attributionsArray = [...(Object.values(attributionsLicenses) as AttributionLicense[])];
 
   const attributions = attributionsArray
```
Comment (SharglutDev) [inline front/src/common/ReleaseInformations/LicenseAttributions.tsx:75]: Should this be in a `useMemo` ?

**Review by SharglutDev — COMMENTED** [review]
Thank you for the PR, left some comments :)

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 7 above)
Comment (Yohh) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx:75]: it's generated at build, don't think so

Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx:75]: I agree with Yoh, a useMemo is not necessary here I think, as the component will not re-render often

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 5 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 6 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done

File: front/src/common/ReleaseInformations/LicenseAttributions.tsx (hunk 5 above)
Comment (clarani) [reply front/src/common/ReleaseInformations/LicenseAttributions.tsx]: Done, good idea :)

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested :)

**Pull request merged**